src/scroll-timeline-css-parser.js

```diff
@@ -0,0 +1,360 @@
+    for (let i = this.cssRulesWithTimelineName.length - 1; i >= 0; i--) {
+      const current = this.cssRulesWithTimelineName[i];
+      if (target.matches(current.selector)) {
+        if (!current['animation-name'] || current['animation-name'] == animationName) {
+          return current['animation-timeline'];
+        }
+      }
+    }
+
+    return null;
+  }
+
+  parseScrollTimeline(p) {
+    const startIndex = p.index;
+    this.assertString(p, "@scroll-timeline");
+    this.eatWhitespace(p);
+    let name = this.parseIdentifier(p);
+    this.eatWhitespace(p);
+    this.assertString(p, "{"); // eats {
+    this.eatWhitespace(p);
+
+    let scrollTimeline = {
+      name: name,
+      source: "auto",
+      orientation: undefined,
+      'scroll-offsets': undefined
+    };
+
+    while (this.peek(p) !== "}") {
+      const property = this.parseIdentifier(p);
+      this.eatWhitespace(p);
+      this.assertString(p, ":");
+      this.eatWhitespace(p);
+      scrollTimeline[property] = this.removeEnclosingDoubleQuotes(this.eatUntil(";", p));
+      this.assertString(p, ";");
+      this.eatWhitespace(p);
+    }
+
+    this.assertString(p, "}");
+    const endIndex = p.index;
+    this.eatWhitespace(p);
+    return {
+      scrollTimeline,
+      startIndex,
+      endIndex,
+    };
+  }
+
+  handleScrollTimelineProps(rule, p) {
+    // TODO is it enough to check with "includes()"
+    const hasAnimationName = rule.block.contents.includes("animation-name:");
+    const hasScrollTimeline = rule.block.contents.includes("animation-timeline:");
+    const hasAnimation = rule.block.contents.includes("animation:");
+
+    // If both 'animation-timeline' and 'animation-name' are present,
+    // save them in the list
+    if (hasScrollTimeline && hasAnimationName) {
+      let timelineNames = RegexMatcher.ANIMATION_TIMELINE
+        .exec(rule.block.contents)?.[1]
+        .trim().split(",").map(name => name.trim());
+
+      let animationNames = RegexMatcher.ANIMATION_NAME
+        .exec(rule.block.contents)?.[1]
+        .trim().split(",").map(name => name.trim());
+
+      for (let i = 0; i < timelineNames.length; i++) {
+        this.cssRulesWithTimelineName.push({
+          selector: rule.selector,
+          'animation-name': animationNames[i],
+          'animation-timeline': timelineNames[i]
+        });
+      }
+      return;
+    }
+
+    let timelineName = RegexMatcher.ANIMATION_TIMELINE
+      .exec(rule.block.contents)?.[1]
+      .trim();
+
+    let animationName = undefined;
+    if (hasAnimationName) {
+      animationName = RegexMatcher.ANIMATION_NAME
+        .exec(rule.block.contents)?.[1]
+        .trim();
+    } else if (hasAnimation) {
+      let shorthand = RegexMatcher.ANIMATION
+        .exec(rule.block.contents)?.[1]
+        .trim();
+
+      if (shorthand) {
+        let remainingTokens = removeKeywordsFromAnimationShorthand(shorthand);
+        // TODO we are assuming the first one that is remaining is
```

Comment:
Should be easy enough to validate the names based on looking them up in @keyframes and @*-timeline rules.